java/arcs/core/data/CapabilityNew.kt

```diff
@@ -131,5 +201,40 @@ sealed class CapabilityNew {
                 else -> Comparison.LessStrict
             }
         }
+
+        companion object {
+            val ANY = Range(Shareable(false), Shareable(true))
+
+            fun fromAnnotations(annotations: List<Annotation>): Shareable? {
+                return annotations.find {
+                    arrayOf("shareable", "tiedToRuntime").contains(it.name)
+                }?.let { CapabilityNew.Shareable(true) }
+            }
+        }
+    }
+
+    data class Range(val min: CapabilityNew, val max: CapabilityNew) : CapabilityNew() {
```

Comment:
Can this be generic class? 

```
class Range<T: CapabilityNew>(val min: T, val max: T): CapabilityNew() {
}
```

Reply:
i guess it could (and then i won't need to verify min and max are of the same type.
i'm having trouble with the `toRange` method's return type declaration though, can't get it to compile with the generic type...

Reply:
Ah, you might need `out` or `in` annotation: I think the code will  get simpler and types more explicit?
https://kotlinlang.org/docs/reference/generics.html#variance

Reply:
i like this, thanks! let me merge this one and do it in a separate follow up pr to unblock other work i'm doing with these files.